app/src/main/java/fr/free/nrw/commons/CommonsApplication.java

```diff
@@ -339,4 +345,97 @@ public interface LogoutListener {
 
         void onLogoutComplete();
     }
+
+    /**
+     * This listener is responsible for handling post-logout actions, specifically invoking the LoginActivity
+     * with relevant intent parameters. It does not perform the actual logout operation.
+     */
+    public static class BaseLogoutListener implements CommonsApplication.LogoutListener {
+
+        Context ctx;
+        String loginMessage, userName;
+
+        /**
+         * Constructor for BaseLogoutListener.
+         *
+         * @param ctx Application context
+         */
+        public BaseLogoutListener(final Context ctx) {
+            this.ctx = ctx;
+        }
+
+        /**
+         * Constructor for BaseLogoutListener
+         *
+         * @param ctx           Application context
+         * @param loginMessage  Message to be displayed on the login page
+         * @param loginUsername Username to be pre-filled on the login page
+         */
+        public BaseLogoutListener(final Context ctx, final String loginMessage,
+            final String loginUsername) {
+            this.ctx = ctx;
+            this.loginMessage = loginMessage;
+            this.userName = loginUsername;
+        }
+
+        @Override
+        public void onLogoutComplete() {
+            Timber.d("Logout complete callback received.");
+            final Intent loginIntent = new Intent(ctx, LoginActivity.class);
+            loginIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
+                .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+            if (loginMessage != null) {
+                loginIntent.putExtra(loginMessageIntentKey, loginMessage);
+            }
+            if (userName != null) {
+                loginIntent.putExtra(loginUsernameIntentKey, userName);
+            }
+
+            ctx.startActivity(loginIntent);
+        }
+    }
+
+    /**
+     * This class is an extension of BaseLogoutListener, providing additional functionality or customization
+     * for the logout process. It includes specific actions to be taken during logout, such as clearing user-specific data
+     * or handling redirection to the login screen.
```

Comment:
> or handling redirection to the login screen.

This is fine.

> such as clearing user-specific data 

I'm not sure how you're mentioning this. This listener is invoked _after_ the user data is cleared right? So, how is it that you claim this listener is responsible for clearing of user-specific data?

Reply:
Sorry , that was a mistake